Source/EventHorizon/Consumer/ConsumerClient.cs

```diff
@@ -141,7 +142,32 @@ protected virtual void Dispose(bool disposeManagedResources)
             _disposed = true;
         }
 
-        async Task<AsyncServerStreamingCall<Contracts.SubscriptionMessage>> Subscribe(SubscriptionId subscriptionId, MicroserviceAddress microserviceAddress)
+        IReverseCallClient<EventHorizonConsumerToProducerMessage, EventHorizonProducerToConsumerMessage, ConsumerSubscriptionRequest, Contracts.SubscriptionResponse, ConsumerRequest, ConsumerResponse> CreateClient(
+            MicroserviceAddress microserviceAddress,
+            CancellationToken cancellationToken)
+        {
+            var client = _clientManager.Get<Contracts.Consumer.ConsumerClient>(
+                microserviceAddress.Host,
+                microserviceAddress.Port);
+            return _reverseCallClients.GetFor<EventHorizonConsumerToProducerMessage, EventHorizonProducerToConsumerMessage, ConsumerSubscriptionRequest, Contracts.SubscriptionResponse, ConsumerRequest, ConsumerResponse>(
+                () => client.Subscribe(cancellationToken: cancellationToken),
+                (message, arguments) => message.SubscriptionRequest = arguments,
+                message => message.SubscriptionResponse,
+                message => message.Request,
+                (message, response) => message.Response = response,
+                (arguments, context) => arguments.CallContext = context,
+                request => request.CallContext,
+                (response, context) => response.CallContext = context,
+                message => message.Ping,
+                (message, pong) => message.Pong = pong,
+                TimeSpan.FromSeconds(7));
```

Comment:
The SDK seems to use 5 seconds for the ping interval, should this also be 5 or is the extra 2 seconds for the server to have time to catch up to?

Reply:
That was the intention. I think it's fair to have a little slower ping interval on event horizon